spark/unitycatalog/src/test/java/com/sparkuctest/UnityCatalogSupport.java

```diff
@@ -0,0 +1,266 @@
+
+  private static final Logger logger = Logger.getLogger(UnityCatalogSupport.class);
+
+  /**
+   * The Unity Catalog server instance.
+   */
+  private UnityCatalogServer ucServer;
+
+  /**
+   * The port on which the UC server is running.
+   */
+  private int ucPort;
+
+  /**
+   * The temporary directory for UC server data.
+   */
+  private File ucTempDir;
+
+  /**
+   * The name of the Unity Catalog in Spark's catalog registry.
+   * Subclasses can override this if they need a different catalog name.
+   */
+  protected String getUnityCatalogName() {
+    return "unity";
+  }
+
+  /**
+   * The URI of the Unity Catalog server.
+   * Available after setup() is called.
+   */
+  protected String getUnityCatalogUri() {
+    return "http://localhost:" + ucPort;
+  }
+
+  /**
+   * The authentication token for the Unity Catalog server.
+   * Currently using a test token; in production scenarios this would be more secure.
+   */
+  protected String getUnityCatalogToken() {
+    return "not-a-token";
+  }
+
+  /**
+   * Creates a Unity Catalog API client configured for this server.
+   */
+  protected ApiClient createUnityCatalogClient() {
+    ApiClient client = new ApiClient();
+    client.setScheme("http");
+    client.setHost("localhost");
+    client.setPort(ucPort);
+    return client;
+  }
+
+  /**
+   * Finds an available port for the UC server.
+   */
+  private int findAvailablePort() throws IOException {
+    try (ServerSocket socket = new ServerSocket(0)) {
+      return socket.getLocalPort();
+    }
+  }
+
+  /**
+   * Starts the Unity Catalog server before all tests.
+   * IMPORTANT: Starts the server BEFORE calling other setup to ensure
+   * the server is running when SharedSparkSession creates the SparkSession.
+   */
+  @Before
+  public void setupUnityCatalog() throws Exception {
+    // Create temporary directory for UC server data
+    ucTempDir = Files.createTempDirectory("unity-catalog-test-").toFile();
+    ucTempDir.deleteOnExit();
+
+    // Find an available port
+    ucPort = findAvailablePort();
+
+    // Set up server properties
+    Properties serverProps = new Properties();
+    serverProps.setProperty("server.env", "test");
+
+    // Start UC server with configuration
+    ServerProperties initServerProperties = new ServerProperties(serverProps);
+
+    UnityCatalogServer server = UnityCatalogServer.builder()
+        .port(ucPort)
+        .serverProperties(initServerProperties)
+        .build();
+
+    server.start();
+    ucServer = server;
+
+    // Poll for server readiness by checking if we can create an API client and query catalogs
+    int maxRetries = 30;
+    int retryDelayMs = 500;
+    boolean serverReady = false;
+    int retries = 0;
+
+    while (!serverReady && retries < maxRetries) {
+      try {
+        ApiClient testClient = new ApiClient();
+        testClient.setScheme("http");
+        testClient.setHost("localhost");
+        testClient.setPort(ucPort);
```

Comment:
Could you please abstract a method to deduplicate the `ApiClient` creation ?

Reply:
There is another place : https://github.com/delta-io/delta/pull/5523/files#diff-745c102864cd5c2f2b62cd6acbb1045f04834be07140567fe8131327e09c64d8R113-R119

Reply:
Yep. Thank you for catching these. I confess that I didn't get a chance to self-review my code.